src/backend/postmaster/pgstat.c

```diff
@@ -2908,11 +2910,19 @@ gpstat_report_waiting(char reason)
 		return;
 
 	/*
-	 * Since this is a single-byte field in a struct that only this process
-	 * may modify, there seems no need to bother with the st_changecount
-	 * protocol.  The update must appear atomic in any case.
+	 * Since we updating multiple fields, go through the st_changecount
+	 * protocol.
```

Comment:
the canonical phrase `protocol of bumping st_changecount before and after` is more relevant here

Reply:
![Webp net-resizeimage](https://user-images.githubusercontent.com/42867699/162230921-30e6679b-7108-4335-a60c-151dd44d6580.jpg)